### PR TITLE
CRI node e2e: add tests for docker 1.10

### PR DESCRIPTION
**Pull request by yujuhong**
This is part of #38164

### CODE REVIEW

**Comment by yujuhong** [comment]
Need to wait until #40946 gets merged.

**Comment by yujuhong** [comment]
The kubelet failed to start up with docker 1.10 in the node e2e:
```
error: failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "cgroupfs" is different from docker cgroup driver: ""
```
I added a new commit to set the default cgroup driver.

File: pkg/kubelet/dockershim/docker_service.go [hunk 1]
```diff
@@ -140,18 +142,21 @@ func NewDockerService(client dockertools.DockerInterface, seccompProfileRoot str
 	glog.Infof("Docker cri networking managed by %v", plug.Name())
 
 	// NOTE: cgroup driver is only detectable in docker 1.11+
-	var cgroupDriver string
+	cgroupDriver := defaultCgroupDriver
 	dockerInfo, err := ds.client.Info()
 	if err != nil {
-		glog.Errorf("failed to execute Info() call to the Docker client: %v", err)
-		glog.Warningf("Using fallback default of cgroupfs as cgroup driver")
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_service.go]: @derekwaynecarr, the code doesn't actually set the driver to cgroupfs. I fix that in this commit.

**Comment by Random-Liu** [comment]
LGTM

**Comment by yujuhong** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot gci gce e2e test this

**Comment by yujuhong** [comment]
@k8s-bot gci gce e2e test this
@k8s-bot kops aws e2e test this

Deployment test flaking...

**Comment by yujuhong** [comment]
Rebased. Applying lgtm based on https://github.com/kubernetes/kubernetes/pull/41092#issuecomment-278199161 and https://github.com/kubernetes/kubernetes/pull/41092#issuecomment-278185263 
Will also check to see if cri node e2e passes.

**Pull request merged**